### ISSUE TITLE
postgresql_extension broken after update

### ISSUE THREAD

**Issue by sauliusgrigaitis** [sous-chefs/postgresql#444]
### Cookbook version
117a795f1af7617c1a462c003406ff32b32665aa

### Chef-client version
12.21.20

### Platform Details
Ubuntu 16.04

### Scenario:
Just updated this cookbook from `245d9d5cb204d6ce8f65890c335baa3b4d02cecd` to `117a795f1af7617c1a462c003406ff32b32665aa` and got this error:

```
 * postgresql_extension[physitrack/plpgsql] action create  * postgresql_extension[physitrack/plpgsql] action create
    
    ================================================================================
    Error executing action `create` on resource 'postgresql_extension[physitrack/plpgsql]'
    ================================================================================
    
    NoMethodError
    -------------
    undefined method `new_resource' for Custom resource postgresql_extension from cookbook postgresql
    
    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/postgresql/resources/extension.rb:63:in `check_extensions_support'
    /var/chef/cache/cookbooks/postgresql/resources/extension.rb:33:in `block in class_from_file'
    
    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/postgresql/recipes/contrib.rb
    
     31:     postgresql_extension "#{db_name}/#{pg_ext}"
     32:   end
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/postgresql/recipes/contrib.rb:31:in `block in from_file'
    
    postgresql_extension("physitrack/plpgsql") do
      action [:create]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :postgresql_extension
      cookbook_name "postgresql"
      recipe_name "contrib"
      database "physitrack"
      extension "plpgsql"
    end
    
    System Info:
    ------------
    chef_version=12.21.20
    platform=ubuntu
    platform_version=16.04
    ruby=ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
    program_name=chef-solo worker: ppid=6061;start=09:02:23;
    executable=/usr/local/bin/chef-solo
```

**Comment by SAPDanJoe** [comment]
It looks like this code was changed with commit b9dd9b2875856d2331451a9ccc840a3f9b05d830 by @damacus  to fix a deprication of the usage in Chef 13.2

I am having trouble using this syntax (`new_resource.property_name`) in my `load_current_value` block with Chef 12.19~

**Comment by SAPDanJoe** [comment]
@damacus - I may have found a solution:
instead of calling `load_current_value do ...`
maybe try `load_current_value do |new_resource|`

**Comment by damacus** [comment]
We're working on a number of resources to make this cookbook easier to maintain

See #454 for progress 👍

**Comment by damacus** [comment]
Relates to #471

**Comment by damacus** [comment]
Relates to #476

**Comment by damacus** [comment]
This should now be fixed by #476 If this isn't the case, pass me the coffee.
